daffodil-macro-lib/src/main/scala/org/apache/daffodil/lib/exceptions/AssertMacros.scala

```diff
@@ -28,7 +28,7 @@ object AssertMacros {
 
     q"""
     if (!($testAbortsIfFalse)) {
-         Assert.abort2("Usage error: " + $message, $testAsString)
+         throw new IllegalStateException("Usage error: " + $message + $testAsString)
```

Comment:
Do we want to remove the testAsString? For an API user, that really isn't going to mean very much. Instead the message should be descriptive enough to tell the user how to fix the issue?